src/main/java/com/segment/analytics/android/integrations/amplitude/AmplitudeIntegration.java

```diff
@@ -177,11 +179,38 @@ public void track(TrackPayload track) {
   }
 
   private void event(
-      @NonNull String name, @NonNull Properties properties, @Nullable JSONObject groups) {
+      @NonNull String name,
+      @NonNull Properties properties,
+      @Nullable Map options,
+      @Nullable JSONObject groups) {
     JSONObject propertiesJSON = properties.toJsonObject();
-    amplitude.logEvent(name, propertiesJSON, groups);
-    logger.verbose(
-        "AmplitudeClient.getInstance().logEvent(%s, %s, %s);", name, propertiesJSON, groups);
+    boolean outOfSession = false;
+    long timestamp = 0;
+
+    if (!isNullOrEmpty(options)) {
+      if (options.containsKey("outOfSession") && options.get("outOfSession") != null) {
+        outOfSession = (Boolean) options.get("outOfSession");
+      }
+      if (options.containsKey("timestamp") && options.get("timestamp") != null) {
+        timestamp = (Long) options.get("timestamp");
+      }
+    }
+
+    if (outOfSession && timestamp > 0) {
+      amplitude.logEvent(name, propertiesJSON, groups, timestamp, true);
+      logger.verbose(
+          "AmplitudeClient.getInstance().logEvent(%s, %s, %s, %s, %s);",
+          name, propertiesJSON, groups, timestamp, true);
+    } else if (outOfSession) {
+      amplitude.logEvent(name, propertiesJSON, groups, true);
+      logger.verbose(
+          "AmplitudeClient.getInstance().logEvent(%s, %s, %s, %s);",
+          name, propertiesJSON, groups, true);
+    } else {
+      amplitude.logEvent(name, propertiesJSON, groups);
```

Comment:
I think if you trace their code, internally probably just calls logEvent with outOfSession = false and timestamp = 0.

So I think instead of having the three branches, we could do the same.

I.e. I think lines 199-213 can be deleted and we could just do:

```

amplitude.logEvent(name, propertiesJSON, groups, timestamp, outOfSession);
```

Reply:
Gotcha. If timestamp is auto-set to 0, the events won't show up in Amplitude's dashboard because their system thinks they were sent in 1970 - I discovered after lots of pain earlier today and a discussion with their support. We could do something like this though, assuming that if the timestamp is greater than zero, the a user has passed in an appropriate timestamp in milliseconds (we'll have to document the format Amplitude expects):

```
if (timestamp > 0) {
  amplitude.logEvent(name, propertiesJSON, groups, timestamp, outOfSession);
} else {
  amplitude.logEvent(name, propertiesJSON, groups, outOfSession);
}
```

Reply:
Looks like instead of 0 it should be current time. 

https://github.com/amplitude/Amplitude-Android/blob/master/src/com/amplitude/api/AmplitudeClient.java#L648

Reply:
Either way, I'm ok with what we have before, reducing to 2 cases like you suggested, or reducing to one case.